internal/controllers/rest_controller.go

```diff
@@ -0,0 +1,78 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/ydataai/authentication-service/internal/clients"
+	"github.com/ydataai/authentication-service/internal/services"
+	"github.com/ydataai/go-core/pkg/common/logging"
+	"github.com/ydataai/go-core/pkg/common/server"
+)
+
+// RESTController defines rest controller
+type RESTController struct {
+	configuration RESTControllerConfiguration
+	sessionConfig services.SessionConfiguration
+	oidcClient    *clients.OIDCClient
+	logger        logging.Logger
+}
+
+// NewRESTController initializes rest controller
+func NewRESTController(
+	configuration RESTControllerConfiguration,
+	sessionConfig services.SessionConfiguration,
+	oidcClient *clients.OIDCClient,
+	logger logging.Logger) RESTController {
+	return RESTController{
+		configuration: configuration,
+		sessionConfig: sessionConfig,
+		oidcClient:    oidcClient,
+		logger:        logger,
+	}
+}
+
+// Boot initialize creating some routes
+func (rc RESTController) Boot(s *server.Server) {
+	s.AddHealthz()
+
+	s.Router.GET(rc.configuration.AuthServiceURL, gin.WrapF(rc.RedirectToOIDCProvider))
+	s.Router.GET(rc.configuration.OIDCCallbackURL, gin.WrapF(rc.OIDCProviderCallback))
+}
+
+// RedirectToOIDCProvider is the handler responsible for redirecting to the OIDC Provider
+func (rc RESTController) RedirectToOIDCProvider(w http.ResponseWriter, r *http.Request) {
+	session, err := services.NewSession(rc.logger, rc.sessionConfig, w, r)
```

Comment:
so, this is initialized in here and then what? services are stateful components, normally, so they should live for all the time of the application, if that really needs to be a service